src/System.Private.CoreLib/shared/System/Security/PermissionSet.cs

```diff
@@ -18,38 +19,38 @@ public partial class PermissionSet : ICollection, IEnumerable, IDeserializationC
         public virtual int Count { get { return 0; } }
         public virtual bool IsReadOnly { get { return false; } }
         public virtual bool IsSynchronized { get { return false; } }
-        public virtual object SyncRoot { get { return null; } }
-        public IPermission AddPermission(IPermission perm) { return AddPermissionImpl(perm); }
-        protected virtual IPermission AddPermissionImpl(IPermission perm) { return default(IPermission); }
+        public virtual object? SyncRoot { get { return null; } }
```

Comment:
Should this rather be changed to `return this` ? It does not make sense for SyncRoot to return null.

Reply:
we were initially asked to minimize IL changes as part of this work - I will send separate PR with this change

Reply:
https://github.com/dotnet/coreclr/pull/23631